### PR TITLE
search: make the number of results configurable

### PR DESCRIPTION
**Pull request by jma**
* Adds a configuration support to customize the number of results text.
* Adds a title on a search view when only one resource is displayed.
* Adds a new configuration to hide the title on the search page.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts [hunk 1]
```diff
@@ -176,7 +201,8 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
   constructor(
     private _recordService: RecordService,
     private _recordUiService: RecordUiService,
-    private _recordSearchService: RecordSearchService
+    private _recordSearchService: RecordSearchService,
+    private _translateService: TranslateService
```
Comment (zannkukai) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.ts:205]: COSMETIC : missing docString

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts [hunk 2]
```diff
@@ -17,12 +17,13 @@
 import { Component, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output, SimpleChanges } from '@angular/core';
 import { FormlyFieldConfig } from '@ngx-formly/core';
 import { JSONSchema7 } from 'json-schema';
-import { Observable, of, Subscription } from 'rxjs';
+import { Observable, of, Subscription, config } from 'rxjs';
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.ts:20]: Not sure that `config` is used.

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts [hunk 3]
```diff
@@ -17,12 +17,13 @@
...
 import { map } from 'rxjs/operators';
 import { ActionStatus } from '../action-status';
 import { RecordUiService } from '../record-ui.service';
 import { RecordService } from '../record.service';
 import { AggregationsFilter, RecordSearchService } from './record-search.service';
+import { TranslateService } from '@ngx-translate/core';
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.ts:26]: Imports are not sorted.

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts [hunk 4]
```diff
@@ -109,8 +110,20 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
...
+    if (this.hits != null && this.hits.hits != null) {
+      return this.hits.hits;
+    }
+    return [];
+  }
+
+  /**
+   * Contain result row data
+   */
+  hits: any = [];
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.ts:126]: getters and setters should be placed after ngOn* methods.

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts [hunk 5]
```diff
@@ -109,8 +110,20 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
 
   /**
    * Contain result data
+   *
+   * @return list of metadata results.
    */
-  records: object[] = [];
+  get records() {
+    if (this.hits != null && this.hits.hits != null) {
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.ts:117]: Can be: 

```
if (this.hits && this.hits.hits)
```

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts [hunk 6]
```diff
@@ -119,8 +132,15 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
 
   /**
    * Total of records corresponding to request
+   *
+   * @return integer representing the number of results.
    */
-  total = 0;
+  get total() {
+    if (this.hits && this.hits.total != null) {
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.ts:139]: Can be: 

```
if (this.hits && this.hits.total)
```

**Pull request merged**